src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -243,11 +245,33 @@ public void AddTagHelperAttribute(string name, object value)
             Output.Reinitialize(tagName, tagMode);
         }
 
+        /// <summary>
+        /// A delegate used to execute children asynchronously with the pages <see cref="HtmlEncoder" /> in scope and
+        /// set <see cref="Output"/>'s <see cref="TagHelperOutput.Content"/> to the rendered results.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that on completion sets <see cref="Output"/>'s
+        /// <see cref="TagHelperOutput.Content"/> to the children's rendered content.</returns>
+        public async Task SetOutputContentAsync()
+        {
+            var childContent = _childContent;
+
+            if (childContent == null)
+            {
+                _startTagHelperWritingScope(null);
+                await _executeChildContentAsync();
+                childContent = _endTagHelperWritingScope();
+            }
+
+            Debug.Assert(!Output.Content.IsModified);
```

Comment:
- Odd to see assertions in a `public` method even though class isn't visible to users.
- Use `Output.IsContentModified` to exactly match the generated code.